doc/dev/custom_devices.rst

```diff
@@ -0,0 +1,159 @@
+            virtual void Counts(DataView<double, 1> &, DataView<int64_t, 1> &, size_t) {}
+
+            virtual void PartialCounts(DataView<double, 1> &, DataView<int64_t, 1> &,
+                                       const std::vector<QubitIdType> &, size_t)
+            {
+            }
+
+            virtual void Gradient(std::vector<DataView<double, 1>> &, const std::vector<size_t> &) {}
+        };
+
+In addition to implementing the ``QuantumDevice`` class, one must implement the following method:
+
+.. code-block:: c++
+
+    extern "C" Catalyst::Runtime::QuantumDevice*
+    getCustomDevice() { return new CustomDevice(); }
+
+where ``CustomDevice()`` is a constructor for your custom device.
+``CustomDevice``'s destructor will be called by the runtime.
+
+.. note::
+
+    This interface might change quickly in the near future.
+    Please check back regularly for updates and to ensure your device is compatible with a specific version of Catalyst.
+
+How to compile custom devices
+=============================
+
+One can follow the ``catalyst/runtime/tests/third_party/CMakeLists.txt`` `as an example. <https://github.com/PennyLaneAI/catalyst/blob/26b412b298f22565fea529d2019554e7ad9b9624/runtime/tests/third_party/CMakeLists.txt>`_
+
+.. code-block:: cmake
+
+        cmake_minimum_required(VERSION 3.20)
+
+        project(third_party_device)
+
+        set(CMAKE_CXX_STANDARD 20)
+        set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
+        add_library(dummy_device SHARED dummy_device.cpp)
+        target_include_directories(dummy_device PUBLIC ${runtime_includes})
+        set_property(TARGET dummy_device PROPERTY POSITION_INDEPENDENT_CODE ON)
+
+
+
+Integration with Python devices
+===============================
+
+If you already have a custom PennyLane device defined in Python and have added a shared object that corresponds to your implementation of the ``QuantumDevice`` class, then all you need to do is to add a ``get_c_interface`` method to your PennyLane device.
+The ``get_c_interface`` method should be a static method that takes no parameters and returns the complete path to your shared library with the ``QuantumDevice`` implementation.
+After doing so, Catalyst should be able to interface with your custom device with no problem.
+
+.. code-block:: python
+
+    class DummyDevice(qml.QubitDevice):
+        """Dummy Device"""
+
+        name = "Dummy Device"
+        short_name = "dummy.device"
+        pennylane_requires = "0.32.0"
+        version = "0.0.1"
+        author = "Erick Ochoa"
+
+        def __init__(self, shots=None, wires=None):
+            super().__init__(wires=wires, shots=shots)
+
+        def apply(self, operations, **kwargs):
+            """Your normal definitions"""
+
+        @staticmethod
+        def get_c_interface():
+            """Location to shared object with C/C++ implementation"""
+            return "/libdummy_device.so"
```

Comment:
@erick-xanadu I have a question -- does this string need to be with respect to a specific path?

Reply:
I have made a check for it to be a full path. But technically other paths (depending on configuration) should also work. Ali and I were just chatting about it. There might be some changes in the near future to this.

Reply:
```suggestion
            return "full/path/to/libdummy_device.so"
```